File: java/leetcode/binarySearch/binarySearch.java

public class binarySearch {

    public int binarySearch(int[] nums, int key) {
        int l = 0, h = nums.length - 1;
        while (l <= h) {
            int m = l + (h-l) / 2;
            if (nums[m] == key) {
                return m;
            } else if (nums[m] > key) {
                h = m-1;
            } else {
                l = m+1;
            }
        }
        return -1;
    }
}

File: golang/leetcode/binarysearch/binary_search.go
package binarysearch

func binarySearch(nums []int, key int) int {
	l, h := 0, len(nums)-1
	var m int
	for l <= h {
		m = l + (h-l)/2
		if nums[m] == key {
			return m
		} else if nums[m] > key {
			h = m - 1
		} else {
			l = m + 1
		}
	}
	return -1
}
